### PR TITLE
[test only] add test for activity list on share expiration.

### PR DESCRIPTION
**Pull request by sakshamgurung**
Issue #968 

Test added for activity list on share expiration.
- user share
- group share
- public link

### CODE REVIEW

File: tests/acceptance/features/apiActivity/list.feature [hunk 1]
```diff
@@ -597,4 +597,88 @@ Feature: List activity
...
+      | subject          | /^shared_with_by$/                                                                                                                                                                   |
+      | object_name      | /^\/lorem.txt$/                                                                                                                                                                      |
+      | object_type      | /^files$/                                                                                                                                                                            |
+      | typeicon         | /^icon-shared$/                                                                                                                                                                      |
+      | subject_prepared | /^<user display-name=\"Alice Hansen\">Alice<\/user> shared <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem.txt" id=\"\d+\">lorem.txt<\/file> with you$/ |
+
+  Scenario: Check activity list after share is expired for group share
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:667]:
```suggestion
  Scenario: Check activity list after share is expired for public link share
```

**Comment by sakshamgurung** [comment]
Activity for expired user share for sharee is shown but not for group sharing and public link.

**Comment by phil-davis** [comment]
> Activity for expired user share for sharee is shown but not for group sharing and public link.

Maybe after the testing app does the "magic"  that expires the share, there should be a step that tries to access the share (test that the sharee cannot see the share). That will ensure that the actual "share delete after expiry" server logic is triggered, which might help to see the activity?

File: tests/acceptance/features/apiActivity/list.feature [hunk 2]
```diff
@@ -597,4 +597,88 @@ Feature: List activity
       | object_name      | /^\/lorem.txt$/       |
       | object_type      | /^files$/             |
       | typeicon         | /^icon-shared$/       |
-      | subject_prepared | /^<user display-name=\"Alice Hansen\">Alice<\/user> shared <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem.txt" id=\"\d+\">lorem.txt<\/file> with you$/|
\ No newline at end of file
+      | subject_prepared | /^<user display-name=\"Alice Hansen\">Alice<\/user> shared <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem.txt" id=\"\d+\">lorem.txt<\/file> with you$/|
+
+  Scenario: Check activity list after share is expired for user share
+    And these users have been created with default attributes and without skeleton files:
```
Comment (Talank) [inline tests/acceptance/features/apiActivity/list.feature:603]:
```suggestion
    Given these users have been created with default attributes and without skeleton files:
```

File: tests/acceptance/features/apiActivity/list.feature [hunk 3]
```diff
@@ -597,4 +597,88 @@ Feature: List activity
...
+      | affecteduser     | /^Brian$/                                                                                                                                   |
+      | app              | /^files_sharing$/                                                                                                                           |
+      | subject          | /^unshared_by$/                                                                                                                             |
+      | object_name      | /^\/lorem.txt$/                                                                                                                             |
+      | object_type      | /^files$/                                                                                                                                   |
+      | subject_prepared | /^The share for <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem\.txt\" id=\"\d+\">lorem\.txt<\/file> expired$/ |
+
+  Scenario: Check activity list after share is expired for group share
+    And these users have been created with default attributes and without skeleton files:
```
Comment (Talank) [inline tests/acceptance/features/apiActivity/list.feature:634]:
```suggestion
    Given these users have been created with default attributes and without skeleton files:
```

File: tests/acceptance/features/apiActivity/list.feature [hunk 4]
```diff
@@ -597,4 +597,88 @@ Feature: List activity
...
+      | object_name      | /^\/lorem.txt$/                                                                                                                                                                      |
+      | object_type      | /^files$/                                                                                                                                                                            |
+      | typeicon         | /^icon-shared$/                                                                                                                                                                      |
+      | subject_prepared | /^<user display-name=\"Alice Hansen\">Alice<\/user> shared <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem.txt" id=\"\d+\">lorem.txt<\/file> with you$/ |
+
+  Scenario: Check activity list after share is expired for public link share
+    And these users have been created with default attributes and without skeleton files:
```
Comment (Talank) [inline tests/acceptance/features/apiActivity/list.feature:668]:
```suggestion
    Given these users have been created with default attributes and without skeleton files:
```

**Review by Talank — COMMENTED** [review]
We do not start the first step of a scenario with an `And` clause

**Comment by phil-davis** [comment]
Note: there is also an existing issue with acceptance test CI, see #975 - so there will be some unrelated fails in this PR until that issue can be fixed.

**Comment by kiranparajuli589** [comment]
lgtm 👍🏼

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm

**Comment by phil-davis** [comment]
drone did not start - no link from this PR to drone. Looking...

File: tests/acceptance/features/apiActivity/list.feature [hunk 5]
```diff
@@ -597,4 +597,91 @@ Feature: List activity
...
+    And group "grp1" has been created
+    And user "Brian" has been added to group "grp1"
+    And user "Alice" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    When user "Alice" creates a share using the sharing API with settings
+      | path       | /lorem.txt |
+      | shareType  | group      |
+      | shareWith  | grp1       |
+      | expireDate | +15 days   |
+    And the administrator expires the last created share using the testing API
+    Then the activity number 1 of user "Alice" should match these properties:
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:648]: Is there an activity number 2 for Alice, that informs Alice that the share expired?

File: tests/acceptance/features/apiActivity/list.feature [hunk 6]
```diff
@@ -597,4 +597,91 @@ Feature: List activity
...
+      | app              | /^files_sharing$/                                                                                                                                                        |
+      | subject          | /^shared_group_self$/                                                                                                                                                    |
+      | object_name      | /^\/lorem.txt$/                                                                                                                                                          |
+      | object_type      | /^files$/                                                                                                                                                                |
+      | subject_prepared | /^You shared <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem\.txt\" id=\"\d+\">lorem\.txt<\/file> with group <parameter>grp1<\/parameter>$/ |
+    And the activity number 1 of user "Brian" should match these properties:
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:657]: Is there an activity number 2 for Brian, that informs Brian that the share expired?

File: tests/acceptance/features/apiActivity/list.feature [hunk 7]
```diff
@@ -597,4 +597,91 @@ Feature: List activity
...
+      | app              | /^files_sharing$/                                                                                                                                                                       |
+      | subject          | /^shared_user_self$/                                                                                                                                                                    |
+      | object_name      | /^\/lorem.txt$/                                                                                                                                                                         |
+      | object_type      | /^files$/                                                                                                                                                                               |
+      | subject_prepared | /^You shared <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/&scrollto=lorem\.txt\" id=\"\d+\">lorem\.txt<\/file> with <user display-name=\"Brian Murphy\">Brian<\/user>$/ |
+    And the activity number 1 of user "Brian" should match these properties:
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:623]: Is there an activity number 1 or 2 for Brian, that informs Brian that the share was shared with him?

File: tests/acceptance/features/apiActivity/list.feature [hunk 8]
```diff
@@ -597,4 +597,91 @@ Feature: List activity
...
+  Scenario: Check activity list after share is expired for public link share
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+    And user "Alice" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    When user "Alice" has created a public link share with settings
+      | path       | /lorem.txt |
+      | expireDate | +15 days   |
+    And the administrator expires the last created share using the testing API
+    Then the activity number 1 of user "Alice" should match these properties:
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:678]: Is there an activity number 2 for Alice, that informs Alice that the public link share expired?

**Review by phil-davis — COMMENTED** [review]
@sakshamgurung drone did not start (I do not know why!) But if you make some changes anyway, it will probably start. Let's see.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm 👍🏼

**Review by phil-davis — APPROVED** [review]

**Pull request merged**